policies/permissions.md

```diff
@@ -0,0 +1,48 @@
+# Permissions across the project
+
+This document describes what access permissions are granted to individuals in
+different [roles in the project](https://www.astropy.org/team.html#roles).
+
+Occasional *temporary* write (or higher) permissions may be granted for specific
+tasks (for example, a maintainer may be given admin rights to a repo to
+configure CI for the first time or similar), be such permissions must be done
+temporarily unless prescribed by this document.
+
+For GitHub, the permissions are enforced by adding indivuals to the GitHub team
+matching their role (For Example, "Astropy Core Maintainers" or "Astropy web
+site maintainers" for a core maintainer or the astropy.org team, respectively),
+a duty primarily performed at the moment by the Coordination Committee.
+Temporary permissions should instead use the "collaborator" feature on Github to
+make it clear that these permissions are temporary in nature.
+
+Additionally, the granter of permissions (usually the Coordination Committee)
+should send a message to the new recipient of write permissions the
+responsibilities and expectations that go with this - a template for this email
+is available [in this repo](../messages/core_write_access.md). That message may
+contain a prompt for a response, which should be cc-ed/forwarded to
+coordinators@astropy.org
+
+## Core package maintainers
+
+All maintainers listed for the core package receive *write access** to the
+repository via the **Astropy Core Maintainers** GitHub team.
+
+## Coordinated package maintainers
+
+Coordinated package maintainers receive **admin access** to the coordinated
+package repositories via the **<package name> maintainers** GitHUb team (e.g.
+'astroquery maintainers').
+
+## Core package release coordinators
+
+Core package release coordinators receive **admin access** to the core
+repository, as well as the astropy-helpers and extension-helpers repositories
+since releases of those packages may be tightly coupled to the core package, as
+well as **write access** to the website repository. This is done via the **Core
+release maintainers** GitHub team.
+
+## Coordination committee
+
+The coordination committee members receive **owner access** to all repositories
+in the astropy organization. In addition, they have access to the project
```

Comment:
I think this is wrong.  The end goal I think should be the efficient and safe operation of the project and the proposed access rights doesn't serve that goal.

HR people on the coco doesn't need to have owner access, in fact, it's a significant security risk if they lack the knowledge and insight of github to have this access. There are plenty of precedents when this approach went wrong (you can blame for not having the written policy to follow, or can be honest).

 On the other hand an operation team need to have the owner access right to be able to e.g. add new people to the wide 'astropy collaborators' team, or to otherwise manage the teams. There are plenty in the project to choose from into this group who are around regularly enough to make the operation efficient.

Reply:
I don't think we should *not* give owner access to the Coco. As the only elected body they should have that authority and we elect general "coco member" not specific roles "coco position 1: code review", "coco position 2: personal", "coco position 3: strategic planning" etc. Thus, I don't think we can pre-select who on the coco does what - we don't even know who gets elected in the future!

If we ever elect someone who is not comfortable handling that they can simply not use the permissions or be educated about what they mean.

On the other hand, that does not mean that the Coco members have to be the *only* people with those permissions, personally, I'm fine with giving owner permissions to e.g. the operation team as well. That should go into a different section though, since this one is named "Coordination committee".

Reply:
Coco members can sit in the relevant group who is responsible for the Github accesses. But it should not come as default as it causes a huge security hole with no operational purpose other than giving power.

Reply:
> simply not use the permissions

which is not how it works, user accounts would not notice if something is only possible because they are owners. So, if you want non-technical people on the coco, then remove the unconditional access by default to all the technical and security side of the project based on coco membership and keep it with the technical team.

Reply:
Now that I read this there's a subtle distinction here.  I'd say that better wording here would be 
the modification I suggested in https://github.com/astropy/astropy-project/pull/148/files#r663198400.  That's not the same as *admin* access to a *repo*, which is important because I think *admin* access should be given more freely than owner*, and should not be particular to the CoCo for the reasons @bsipocz says.

But assuming that clarification is made, it's instructive to look at this list to see what owner gives: https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/permission-levels-for-an-organization .

That includes a *lot* of power that can 1) be abused and 2) sometimes needs to be used (responsibly).  And while some of it is highly technical (e.g., code review settings and github apps), other parts are not, but do require a lot of care from a *policy*/social perspective 
(e.g. managing comments that are disruptive or violate the CoC, keeping track of who should be on a team because of their role in the Project). This means it can't be given out freely with exactly the goal @bsipocz said (that I wholly agree with):

> The end goal I think should be the efficient and safe operation of the project

But as @hamogu said, the whole point is that the group that APE0 says this authority is entrusted  to is the CoCo, by virtue of the election process.  Now in principle the CoCo can delegate some of that power, which is exactly why people have gotten admin or maintainer powers in lots of places.  So if it's not the CoCo, who is it that should get every single one of those powers, and how is that group decided on? I'd say the whole point of the CoCo is to be the body to do that, but only as long as the community *as a whole* (not any individual) entrusts them with that.

To emphasize, I think it's a fair question whether people *not* in the CoCo should also get this power.  There have been times when it made sense to add someone when they needed to exercise these powers, and then back them down to "standard" privledges as soon as they'd done this. Bbut I think APE0 pretty clearly says it's the CoCo who ave the authority to control this power, and mechanically the way they control who gets it is by having it themslves (since only owners can add/remove other owners).

Reply:
> (e.g. managing comments that are disruptive or violate the CoC, keeping track of who should be on a team because of their role in the Project)

managing comments are not a daily task the CoCo does. I bet none on the coco follows the comments anyway, they are notified of incidents if any.

> of care from a policy/social perspective
> That includes a lot of power that can 1) be abused and

Let's not go into this direction. One person has a track record of deleting emails from email lists and abusing power because they think so on their own and because they have the power. Keeping even more technical power including to delete anything in the org is a proven security hazard on top of things.

Reply:
And to reiterate: "to ensure the efficient, safe, and secure operation of the project", technical keys should not be with people who don't know the technical side of github operation. It's unfortunate that some of the HR keys are also stored with the same access, but I firmly believe that the direction should go the other way. The technical team should add temporarily access to moderate when a situation arises and not everyone on CoCo by detail to keep the tech keys and ability to delete the project, opt into services etc.